app/src/main/java/de/westnordost/streetcomplete/data/osmtracks/TracksApiImpl.kt

```diff
@@ -0,0 +1,70 @@
+package de.westnordost.streetcomplete.data.osmtracks
+
+import de.westnordost.osmapi.OsmConnection
+import de.westnordost.osmapi.common.errors.OsmApiException
+import de.westnordost.osmapi.common.errors.OsmApiReadResponseException
+import de.westnordost.osmapi.common.errors.OsmAuthorizationException
+import de.westnordost.osmapi.common.errors.OsmConnectionException
+import de.westnordost.osmapi.map.data.OsmLatLon
+import de.westnordost.osmapi.traces.GpsTraceDetails
+import de.westnordost.osmapi.traces.GpsTracesApi
+import de.westnordost.osmapi.traces.GpsTrackpoint
+import de.westnordost.streetcomplete.ApplicationConstants
+import de.westnordost.streetcomplete.data.download.ConnectionException
+import de.westnordost.streetcomplete.data.user.AuthorizationTracesException
+import java.time.Instant
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+
+class TracksApiImpl(osm: OsmConnection) : TracksApi {
+    private val api: GpsTracesApi = GpsTracesApi(osm)
+
+    override fun create(trackpoints: List<Trackpoint>, noteText: String?): Track = wrapExceptions {
+
+        // Filename is just the start of the track
+        // https://stackoverflow.com/a/49862573/7718197
+        val name = DateTimeFormatter
+            .ofPattern("yyyy_MM_dd'T'HH_mm_ss.SSSSSS'Z'")
+            .withZone(ZoneOffset.UTC)
+            .format(Instant.ofEpochSecond(trackpoints[0].time)) + ".gpx"
+        val visibility = GpsTraceDetails.Visibility.IDENTIFIABLE
+        val description = if (noteText != null && noteText.isNotBlank()) {
+            noteText
+        } else {
+            "Uploaded via " + ApplicationConstants.USER_AGENT
+        }
```

Comment:
```suggestion
        val description = if (noteText.isNullOrEmpty()) "Uploaded via ${ApplicationConstants.USER_AGENT}" else noteText
```

Reply:
Note: This changes the semantics, because `blank` and `empty` are different. (blank also includes any whitespace)

Here is the shortest/"cleanest" version I came up with
```suggestion
        val description = noteText.orEmpty().ifBlank { "Uploaded via ${ApplicationConstants.USER_AGENT}" }
```

Other options:
```kt
val description = noteText?.takeIf { !it.isBlank() } ?: "Uploaded via ${ApplicationConstants.USER_AGENT}"
val description = noteText?.takeUnless { it.isBlank() } ?: "Uploaded via ${ApplicationConstants.USER_AGENT}"
```

…Or don't change anything. Tbh, recently I've been falling out of love with kotlin's stdlib. I like the readability of the code it produces in any given spot, but it is bloat which really increases the mental overhead of the language in general. :shrug: